### ISSUE TITLE
HadamardPower(A, B).as_explicit() fails

### ISSUE THREAD

**Issue by cbm755** [sympy/sympy#16505]
Matrix to matrix power fails:

```python
>>> A = MatrixSymbol('A', 2, 3)
>>> HadamardPower(A, A).as_explicit()
---------------------------------------------------------------------------

NotImplementedError: Matrix Power not defined
```

**Comment by sachin-4099** [comment]
Want to take up this issue if it is still open. Please guide me as to how to proceed. Which part of the code needs to be changed? What are the testcases to be added?
@oscarbenjamin @sylee957

**Comment by oscarbenjamin** [comment]
The current error message is:
```python
>>> A = MatrixSymbol('A', 2, 3)
>>> HadamardPower(A, A).as_explicit()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/enojb/current/sympy/sympy/sympy/matrices/expressions/matexpr.py", line 340, in as_explicit
    for i in range(self.rows)])
  File "/Users/enojb/current/sympy/sympy/sympy/matrices/expressions/matexpr.py", line 340, in <listcomp>
    for i in range(self.rows)])
  File "/Users/enojb/current/sympy/sympy/sympy/matrices/expressions/matexpr.py", line 339, in <listcomp>
    for j in range(self.cols)]
  File "/Users/enojb/current/sympy/sympy/sympy/matrices/expressions/matexpr.py", line 289, in __getitem__
    return self._entry(i, j)
  File "/Users/enojb/current/sympy/sympy/sympy/matrices/expressions/hadamard.py", line 320, in _entry
    return self.base._entry(i, j, **kwargs)**self.exp
  File "/Users/enojb/current/sympy/sympy/sympy/core/expr.py", line 210, in __pow__
    return self._pow(other)
  File "/Users/enojb/current/sympy/sympy/sympy/core/decorators.py", line 91, in __sympifyit_wrapper
    return func(a, b)
  File "/Users/enojb/current/sympy/sympy/sympy/core/decorators.py", line 128, in binary_op_wrapper
    return f(self)
  File "/Users/enojb/current/sympy/sympy/sympy/matrices/expressions/matexpr.py", line 24, in __sympifyit_wrapper
    return func(a, b)
  File "/Users/enojb/current/sympy/sympy/sympy/core/decorators.py", line 129, in binary_op_wrapper
    return func(self, other)
  File "/Users/enojb/current/sympy/sympy/sympy/matrices/expressions/matexpr.py", line 141, in __rpow__
    raise NotImplementedError("Matrix Power not defined")
NotImplementedError: Matrix Power not defined
```
So I would start by looking around line 141 in matexpr.py.

**Comment by sylee957** [comment]
I already have a fix in #16508
